tests/foreman/cli/test_partitiontable.py

```diff
@@ -74,49 +82,102 @@ def test_positive_delete_by_id(self):
+        ptable = make_partition_table()
+        PartitionTable.delete({'id': ptable['id']})
+        with self.assertRaises(CLIReturnCodeError):
+            PartitionTable.info({'id': ptable['id']})
+
+    @tier1
+    def test_positive_delete_by_name(self):
+        """Create a Partition Table then delete it by its name
+
+        @Feature: Partition Table
+
+        @Assert: Partition Table is deleted
+        """
+        ptable = make_partition_table()
+        PartitionTable.delete({'name': ptable['name']})
+        with self.assertRaises(CLIReturnCodeError):
+            PartitionTable.info({'name': ptable['name']})
 
-    # pylint: disable=no-self-use
     @tier2
     def test_positive_add_os_by_id(self):
-        """Create a partition table then add an operating system to it
+        """Create a partition table then add an operating system to it using
+        IDs for association
 
         @Feature: Partition Table
 
         @Assert: Operating system is added to partition table
-
         """
-        content = "Fake ptable"
-        ptable = make_partition_table({'content': content})
-        os_list = OperatingSys.list()
-        PartitionTable().add_operating_system({
+        ptable = make_partition_table()
+        os = make_os()
+        PartitionTable.add_operating_system({
             'id': ptable['id'],
-            'operatingsystem-id': os_list[0]['id']
+            'operatingsystem-id': os['id'],
         })
+        ptable = PartitionTable.info({'id': ptable['id']})
+        self.assertIn(os['title'], ptable['operating-systems'])
+
+    @tier2
+    def test_positive_add_os_by_name(self):
+        """Create a partition table then add an operating system to it using
+        names for association
+
+        @Feature: Partition Table
+
+        @Assert: Operating system is added to partition table
+        """
+        ptable = make_partition_table()
+        os = make_os()
+        PartitionTable.add_operating_system({
+            'name': ptable['name'],
+            'operatingsystem': os['title'],
+        })
+        ptable = PartitionTable.info({'name': ptable['name']})
+        self.assertIn(os['title'], ptable['operating-systems'])
 
     @tier2
     def test_positive_remove_os_by_id(self):
-        """Add an operating system to a partition table then remove it
+        """Add an operating system to a partition table then remove it. Use IDs
+        for removal
 
         @Feature: Partition Table
 
         @Assert: Operating system is added then removed from partition table
         """
-        ptable = make_partition_table({'content': gen_string("alpha", 10)})
+        ptable = make_partition_table()
         os = make_os()
         PartitionTable.add_operating_system({
             'id': ptable['id'],
             'operatingsystem-id': os['id'],
         })
         ptable = PartitionTable.info({'id': ptable['id']})
-        self.assertIn(os['title'], ptable['operating-systems'])
```

Comment:
And one more thing - we have `test_positive_add_os_***` tests, which will fail in case addition is unsuccessful, that assertion is there in place so we won't miss such bug.